cms/djangoapps/contentstore/docs/decisions/0002-restful-studio-apis-for-MFE.rst

```diff
@@ -0,0 +1,64 @@
+0002: Expose Existing Studio APIs for use in MFEs
+=================================================
+
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+New MFEs in in development that need to use studio APIs to interact with course tabs
+and advanced course settings. Currently these endpoints don't have RESTful APIs that
+can be used by MFEs using OAuth2 or similar.
+
+Currently the course tabs page at `<studio_url>/tabs/<course_key>` has a single
+`old-style functional view`_ that serves as both the view for the HTML page and the
+JSON-based API for post requests. It also only supports session-based auth. While
+this works well currently, for MFE consumption this is not ideal. The
+`advanced course settings`_ view is in a similar situation.
+
+.. _old-style functional view: https://github.com/edx/edx-platform/blob/49296005db7397e1a45e2864d93d39cf790a5fce/cms/djangoapps/contentstore/views/tabs.py#L27
+.. _advanced course settings: https://github.com/edx/edx-platform/blob/49296005db7397e1a45e2864d93d39cf790a5fce/cms/djangoapps/contentstore/views/course.py#L1367
+
+To avoid disrupting the existing studio views that might still be needed for the next
+few releases, we need to maintain these existing views while also enabling MFEs to
+perform operations currently only enabled by these views.
+
+Decision
+--------
+
+We will expose the existing studio APIs under new a new versioned endpoint. These
+endpoints will support the new JWT/OAuth2 authentication mechanisms supported by
+MFEs.
```

Comment:
Do we already have JWT/Oauth2 endpoints in the studio? Can we mention that here? Or if we have to build one then probably endpoints can be listed here?

Reply:
There may be some, but that doesn't seem relevant here since it's about other endpoints that don't support OAuth2.

Reply:
@xitij2000 Makes sense. 👍🏽